### PR TITLE
Fix #63183 Pods on different nodes mount Ceph RBD PVC stuck on ContainerCreating

### LINKED ISSUE #63183: Pods on different nodes mount Ceph RBD PVC stuck on ContainerCreating
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
When use ceph as storageclass, i set accessModes:  ReadOnlyMany, and create pods using the claim, only one pod would be running and the others on different nodes would get stuck on ContainerCreating. But if the pods are all on same node, it works fine. 

**What you expected to happen**:
Pods on different nodes can mount ceph rbd volume.

**How to reproduce it (as minimally and precisely as possible)**:

```
# cat pvc.yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: ceph-claim
spec:
  accessModes:
    - ReadOnlyMany
  resources:
    requests:
      storage: 2Gi
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.9.2
- Cloud provider or hardware configuration: 
- OS (e.g. from /etc/os-release): 3.10.0-327.18.2.el7.x86_64
- Kernel (e.g. `uname -a`): CentOS Linux release 7.2.1511 (Core)
- Install tools: kubeadm
- Others:

### PR DESCRIPTION
**Pull request by xiaoxubeii**
**What this PR does / why we need it**:
Ceph/RBD pvc can be attached on different nodes when use `ReadOnlyMany`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63183

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/volume/rbd/rbd.go [hunk 1]
```diff
@@ -254,13 +254,16 @@ func (plugin *rbdPlugin) createMounterFromVolumeSpecAndPod(spec *volume.Spec, po
 		}
 	}
 
+	am, err := getVolumeAccessMode(spec)
+
```
Comment (dixudx) [inline pkg/volume/rbd/rbd.go]: No error handling here?

File: pkg/volume/rbd/rbd.go [hunk 2]
```diff
@@ -320,6 +323,8 @@ func (plugin *rbdPlugin) newMounterInternal(spec *volume.Spec, podUID types.UID,
 		return nil, err
 	}
 
+	am, err := getVolumeAccessMode(spec)
+
```
Comment (dixudx) [inline pkg/volume/rbd/rbd.go]: Ditto

File: pkg/volume/rbd/rbd.go [hunk 3]
```diff
@@ -729,6 +736,13 @@ func getVolumeSourceReadOnly(spec *volume.Spec) (bool, error) {
 	return false, fmt.Errorf("Spec does not reference a RBD volume type")
 }
 
+func getVolumeAccessMode(spec *volume.Spec) ([]v1.PersistentVolumeAccessMode, error) {
+	if spec.PersistentVolume != nil {
```
Comment (dixudx) [inline pkg/volume/rbd/rbd.go]: We need also to check the existence of rbd volume.

```
if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.RBD != nil
```

Comment (dixudx) [reply pkg/volume/rbd/rbd.go]: And return error for non RBD volume resources.

File: pkg/volume/rbd/rbd_util.go [hunk 4]
```diff
@@ -269,7 +269,18 @@ func (util *RBDUtil) AttachDisk(b rbdMounter) (string, error) {
 		if err != nil {
 			return "", fmt.Errorf("error: %v, rbd output: %v", err, rbdOutput)
 		}
-		if found {
+
+		// If accessModes contain ReadOnlyMany, we don't need check rbd status of being used.
+		needValidFound := true
+		if b.accessModes != nil {
+			for _, v := range b.accessModes {
+				if v == v1.ReadOnlyMany {
```
Comment (dixudx) [inline pkg/volume/rbd/rbd_util.go]: Shall we use `v != v1.ReadWriteOnce` instead to make better clarification.
Although `ReadWriteMany` is not allowed for RBD volumes.

**Review by dixudx — CHANGES_REQUESTED** [review]

**Comment by dixudx** [comment]
@xiaoxubeii Currently this PR is targeting `release-1.9`. This is not what k/k actually does.

I suggest changing to `master` and cherry-picking to `release-1.9` when needed.

File: pkg/volume/rbd/rbd_util.go (hunk 4 above)
Comment (xiaoxubeii) [reply pkg/volume/rbd/rbd_util.go]: That't right. I will correct it.

**Comment by xiaoxubeii** [comment]
OK, i will pr to master. Thanks : )

**Comment by dixudx** [comment]
@xiaoxubeii No need to close this PR and open a new one.

You can follow [this guide](https://blog.github.com/2016-08-15-change-the-base-branch-of-a-pull-request/) to change the branch. You may also need to rebase it afterwards.

**Pull request closed without merging**